notify/impl.go

```diff
@@ -227,38 +235,38 @@ func (n *Email) auth(mechs string) (smtp.Auth, error) {
 }
 
 // Notify implements the Notifier interface.
-func (n *Email) Notify(ctx context.Context, as ...*types.Alert) error {
+func (n *Email) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
 	// Connect to the SMTP smarthost.
 	c, err := smtp.Dial(n.conf.Smarthost)
 	if err != nil {
-		return err
+		return true, err
 	}
 	defer c.Quit()
 
 	// We need to know the hostname for both auth and TLS.
 	host, _, err := net.SplitHostPort(n.conf.Smarthost)
 	if err != nil {
-		return fmt.Errorf("invalid address: %s", err)
+		return false, fmt.Errorf("invalid address: %s", err)
 	}
 
 	if n.conf.RequireTLS {
 		if ok, _ := c.Extension("STARTTLS"); !ok {
-			return fmt.Errorf("require_tls: true (default), but %q does not advertise the STARTTLS extension", n.conf.Smarthost)
+			return false, fmt.Errorf("require_tls: true (default), but %q does not advertise the STARTTLS extension", n.conf.Smarthost)
 		}
 		tlsConf := &tls.Config{ServerName: host}
 		if err := c.StartTLS(tlsConf); err != nil {
-			return fmt.Errorf("starttls failed: %s", err)
+			return false, fmt.Errorf("starttls failed: %s", err)
```

Comment:
True, then I'll make all request/response calls retry. I missed that and only thought of "is a TLS likely to be recovered on a retry?".